compiler/rustc_mir_transform/src/promote_consts.rs

```diff
@@ -83,23 +84,21 @@ enum TempState {
     PromotedOut,
 }
 
-/// A "root candidate" for promotion, which will become the
-/// returned value in a promoted MIR, unless it's a subset
-/// of a larger candidate.
-#[derive(Copy, Clone, PartialEq, Eq, Debug)]
-struct Candidate {
-    location: Location,
-}
+/// Each `[Location]` is a "root candidate" for promotion, which will become
+/// the returned value in a promoted MIR, unless it's a subset of a larger
+/// candidate.
+#[derive(Clone, Debug)]
+struct Candidates(Vec<Location>);
```

Comment:
I'm not convinced by this change. `Candidate` is a newtype to encode the invariant that the `Location` points to a "root candidate". Using a `Location` everywhere defeats that purpose.

Reply:
It makes sense. Reverted!